_posts/2024-06-03-kasten-and-strimzi-kafka.markdown

```diff
@@ -0,0 +1,241 @@
+---
+layout: post
+title: Protecting Strimzi Kafka with Kasten
+description: How to backup and restore kafka messaging data & config using Kasten by Veeam
+date: 2024-06-03 10:48:00 +0000
+author: jamestate
+image: '/images/posts/2024-06-03-kasten-and-strimzi-kafka/kafka.png'
+image_caption: 'How to protect kafka with kasten'
+tags: [strimzi, kafka, kasten, messaging]
+featured: false
+---
+
+## Kafka Data Protection & The Problem
+
+Kafka is a very popular platform for data flow and pipelining within the enterprise space, especially in Kubernetes environments, where data needs to be shared often between namespaces, pods, applications and clusters. It does pose challenges when assessing it's data protection needs. 
+
+Whilst most of the data is transient, being governed by a timeout removal schedule, the issue is there is no formal backup procedure for Kafka. In fact even Confluence, Kafka's vendor, does not provide a tool for native backup. This doesn't mean it's not important to protect the data. The 'normal' procedure would be to protect the data upstream and downstream from the messaging queue, usually in other applications, but not the data in transmission. However there maybe scenarios where recovery of this transient data is required, and as long as recovery is done before the data expiry time window closes, that data can be recovered. 
+
+From a Kasten perspective it's just another classical Kubernetes application, however the Kafka architecture does require some finesse in restoration. Firstly it's important to understand what Kafka is and the components.  
+
+
+***
+
+## Strimzi
+
+[Strimzi](https://strimzi.io/) is a popular operator in the Kubernetes space, that streamlines and simplifies the deployment of Kafka clusters. It's fairly simple in it's operating model and provides a useful management of Kafka clusters at scale:
+
+*  The operator is installed via HELM in it's own namespace and is configured to watch ALL namespaces for Kafka configurations (via a CRD).
+*  When the operator finds a namespace with a Kafka configuration, it will deploy that configuration into the end namespace and manage the spinning up of pods, creation of PVCs and creation of all services.
+
+As such it's actually very easy to build and deploy multiple Kafka clusters in a very quick fashion. In the below example we will deploy a simple Strimzi Kafka cluster, populate some sample data and backup/restore it.
+
+
+***
+
+
+## Deploying Strimzi & Creating a Kafka Cluster
+
+Assuming we have a working cluster with a default storageclass, HELM installed and kubectrl access to the cluster we can go ahead and install the Strimzi operator:
```

Comment:
Assuming we have a working cluster with a default storageClass, HELM installed and kubectrl access to the cluster, let's install the Strimzi operator: